src/core/messages.ts

```diff
@@ -167,6 +155,11 @@ export type MessageListener = (event: MessageEvent) => void;
 export interface MessageSubscriptionResponse extends Subscription {
   /**
    * Get the previous messages that were sent to the room before the listener was subscribed.
+   *
+   * If you are subscribed to and notified of a discontinuity event, this will reset the starting point of
```

Comment:
Might just be me reading this, but it sort of reads like you have to be subscribed and notified of a discontinuity for the resetting here to take place. This happens regardless of if you have a listener subscribed to discontinuity events though.

What about something like..
```suggestion
* If the client experiences a period of discontinuity, the `getPreviousMessages` starting point will be reset.
* Calls to `getPreviousMessages` during this period will wait for discontinuity to be restored before resolving.
* When continuity is restored, the starting point will be set to the beginning of this new period of continuity.
* As such, to ensure no messages are missing, you should re-call `getPreviousMessages` after any period of discontinuity.
```

Reply:
5b22df6364ebf2886f7a56a4cc250a40f3f59922